pkg/queue/handlers/http_request.go

```diff
@@ -0,0 +1,258 @@
+// JSON objects, otherwise the task will fail.
+func NewHTTPRequestHandler(tokenHeaderName string, tokenCreator tokens.Creator, client *http.Client) workers.TaskHandler {
+	return &httpRequestHandler{
+		Tracer:          tracing.NewTracer("handlers", "HTTPRequestHandler"),
+		tokenHeaderName: tokenHeaderName,
+		tokenCreator:    tokenCreator,
+		client:          client,
+	}
+}
+
+type httpRequestHandler struct {
+	tracing.Tracer
+	tokenHeaderName string
+	tokenCreator    tokens.Creator
+	client          *http.Client
+}
+
+func (h *httpRequestHandler) Process(ctx context.Context, task queue.Task, heartbeats chan<- queue.Progress) (err error) {
+	span, ctx := h.StartSpan(ctx, "Process")
+	defer func() {
+		close(heartbeats)
+		h.FinishSpan(span, err)
+	}()
+	span.SetTag("task.id", task.ID)
+	span.SetTag("task.queue", task.Queue)
+	span.SetTag("task.type", task.Type)
+	span.SetTag("task.spec", string(task.Spec))
+
+	logrus := logrus.WithField("type", task.Type).WithField("id", task.ID)
+
+	logrus.Debug("starting the HTTP request task...")
+
+	var progress HTTPRequestProgress
+	defer func() {
+		// we check for errSerializingHearbeat so we don't cause
+		// a recursion call
+		if err == nil || err == ErrSerializingHearbeat {
+			return
+		}
+		message := err.Error()
+		progress.ErrorMessage = &message
+		_ = sendHTTPRequestProgress(progress, heartbeats)
+	}()
+
+	var spec HTTPRequestTaskSpec
+	err = json.Unmarshal(task.Spec, &spec)
+	if err != nil {
+		return err
+	}
+
+	progress.Stage = RequestPreparing
+	err = sendHTTPRequestProgress(progress, heartbeats)
+	if err != nil {
+		return err
+	}
+
+	var payload io.Reader
+	if spec.RequestBody != "" {
+		payload = strings.NewReader(spec.RequestBody)
+	}
+
+	req, err := http.NewRequest(spec.Method, spec.URL, payload)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("User-Agent", "Contiamo Hub HTTP Request Task")
+
+	for name, value := range spec.RequestHeaders {
+		req.Header.Add(name, value)
+	}
+
+	if spec.Authorized {
+		token, err := h.tokenCreator.Create("httpRequestTask")
+		if err != nil {
+			return err
+		}
+
+		req.Header.Add(h.tokenHeaderName, token)
+	}
+
+	err = opentracing.GlobalTracer().Inject(
+		span.Context(),
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(req.Header),
+	)
+	if err != nil {
+		otext.Error.Set(span, true)
+		span.SetTag("tracing.inject.err", err.Error())
+		err = nil
+	}
+
+	progress.Stage = RequestPending
+	err = sendHTTPRequestProgress(progress, heartbeats)
+	if err != nil {
+		return err
+	}
+
+	now := time.Now()
+	defer func() {
+		duration := time.Since(now).Milliseconds()
+		progress.Duration = &duration
+		err := sendHTTPRequestProgress(progress, heartbeats)
+		if err != nil {
+			logrus.Error(err)
+		}
+	}()
+
+	resp, err := h.client.Do(req.WithContext(ctx))
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	contentType := resp.Header.Get("content-type")
+	if !strings.Contains(contentType, "application/json") {
```

Comment:
why is this check needed?

Reply:
Please read the implementation to the end. We support partial results which are new-line separated JSON objects, this task handlers supports ONLY JSON responses.